File: TypeScript/0014-longest-common-prefix.ts
function longestCommonPrefix(strs: string[]): string {
    if (strs.length === 0) return "";

    let result = "";

    for (let j = 0; j < strs[0].length; j++) {
        const char = strs[0][j];

        for (let i = 1; i < strs.length; i++) {
            if (j >= strs[i].length || strs[i][j] !== char) {
                return result;
            }
        }

        result += char;
    }

    return result;
}

File: Python/0014-longest-common-prefix.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        result = ""

        for j in range(len(strs[0])):
            char = strs[0][j]
            for i in range(1, len(strs)):
                if j >= len(strs[i]) or strs[i][j] != char:
                    return result
            result += char

        return result
